cai_causal_graph/time_series_causal_graph.py

```diff
@@ -723,6 +724,9 @@ def add_node(
         identifier = self._check_node_exists(node)
         self._nodes_by_identifier[identifier] = node
 
+        if all(hasattr(self, attr) for attr in ['lag_to_nodes', 'variable_name_to_nodes']):
+            self.update_cache(node)
```

Comment:
why check for the `hasattr`?

Reply:
Was calling `add_node` from `Causal_Graph` not `TimeSeriesCausalGraph`, `CausalGraph` did not have these attributes. This has been removed now!